contrib/extensions/sling-oak-restrictions/src/main/java/org/apache/sling/slingoakrestrictions/impl/SlingRestrictionProviderImpl.java

```diff
@@ -0,0 +1,107 @@
+import org.apache.jackrabbit.oak.spi.security.authorization.restriction.RestrictionProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+
+/** Sling restriction provider implementation that supports the following restrictions:
+ *
+ * <ul>
+ * <li>{@link #REP_SLING_RESOURCE_TYPE}: A restriction that allows to match against resource types.</li>
+ * </ul>
+ * 
+ * Further sling restriction can be added here in future.
+*/
+@Component
+@Service(RestrictionProvider.class)
+public class SlingRestrictionProviderImpl extends AbstractRestrictionProvider {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SlingRestrictionProviderImpl.class);
+
+    public static final String REP_SLING_RESOURCE_TYPE = "rep:slingResourceTypes";
+
+    public SlingRestrictionProviderImpl() {
+        super(supportedRestrictions());
+    }
+
+    private static Map<String, RestrictionDefinition> supportedRestrictions() {
+        RestrictionDefinition slingResourceTypes = new RestrictionDefinitionImpl(REP_SLING_RESOURCE_TYPE, Type.STRINGS, false);
+        return ImmutableMap.of(slingResourceTypes.getName(), slingResourceTypes);
+    }
+
+    // ------------------------------------------------< RestrictionProvider >---
+
+    @Nonnull
+    @Override
+    public RestrictionPattern getPattern(String oakPath, @Nonnull Tree tree) {
+        if (oakPath != null) {
+            PropertyState resourceTypes = tree.getProperty(REP_SLING_RESOURCE_TYPE);
+            if (resourceTypes != null) {
+                ResourceTypePattern resourceTypePattern = new ResourceTypePattern(resourceTypes.getValue(Type.STRINGS));
+                LOG.debug("Returning resourceTypePattern={} in getPattern(String oakPath, @Nonnull Tree tree) ", resourceTypePattern);
+                return resourceTypePattern;
+            }
+        }
+        return RestrictionPattern.EMPTY;
+    }
+
+    @Nonnull
+    @Override
+    public RestrictionPattern getPattern(@Nullable String oakPath, @Nonnull Set<Restriction> restrictions) {
+
+        if (oakPath != null && !restrictions.isEmpty()) {
+            for (Restriction r : restrictions) {
+                String name = r.getDefinition().getName();
+                if (REP_SLING_RESOURCE_TYPE.equals(name)) {
+                    ResourceTypePattern resourceTypePattern = new ResourceTypePattern(r.getProperty().getValue(Type.STRINGS));
+                    LOG.debug(
+                            "Returning resourceTypePattern={} in getPattern(@Nullable String oakPath, @Nonnull Set<Restriction> restrictions)",
+                            resourceTypePattern);
+                    return resourceTypePattern;
+                }
+            }
+        }
+
+        return RestrictionPattern.EMPTY;
+    }
+
+    @Override
+    public void validateRestrictions(String oakPath, @Nonnull Tree aceTree) throws AccessControlException {
```

Comment:
My point was that you're overriding the method just to call the superclass method. Don't you achieve the same effect my simply removing the method?